### PR TITLE
[tests-only] adding test for sharedWithMe end point for resource in project space and personal space

### PR DESCRIPTION
**Pull request by ghost**
## Description

Scenarios added :
```feature
Scenario: user lists the file shared with them from project space
Scenario: user lists the folder shared with them from project space
Scenario: user lists the folder shared with them from project space after the sharer is disabled
Scenario: user lists the folder shared with them from project space after the sharer is deleted
Scenario: user lists the file shared with them from personal space after the sharer is deleted
```

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/7826

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 1]
```diff
@@ -474,3 +486,1022 @@ Feature: an user gets the resources shared to them
...
+                  "user"
+                ],
+                "properties": {
+                  "user": {
+                    "type": "object",
+                    "required": ["displayName", "id"],
+                    "properties": {
+                      "displayName": {
+                        "type": "string",
+                        "enum": [""]
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature:540]: Yeah, it should be. Please, raise an issue

CC @ScharfViktor

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 2]
```diff
@@ -14,7 +14,13 @@ Feature: an user gets the resources shared to them
...
   Scenario: user lists the file shared with them
     Given user "Alice" has uploaded file with content "hello world" to "/textfile0.txt"
-    And user "Alice" has shared file "textfile0.txt" with user "Brian"
+    And user "Alice" has sent the following share invitation:
+      | resourceType    | file          |
+      | resource        | textfile0.txt |
+      | space           | Personal      |
+      | sharee          | Brian         |
+      | shareType       | user          |
+      | permissionsRole | Viewer        |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:23]: could you move this to another PR and then we can merge it

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 3]
```diff
@@ -484,3 +484,1020 @@ Feature: an user gets the resources shared to them
...
+    Given the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created a space "new-space" with the default quota using the Graph API
+    And user "Alice" has created a folder "folder" in space "new-space"
+    And user "Alice" has sent the following share invitation:
+      | resourceType    | folder     |
+      | resource        | folder |
+      | space           | new-space  |
+      | sharee          | Brian      |
+      | shareType       | user       |
+      | permissionsRole | Viewer     |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:768]:
```suggestion
      | resourceType    | folder     |
      | resource        | folder     |
      | space           | new-space  |
      | sharee          | Brian      |
      | shareType       | user       |
      | permissionsRole | Viewer     |
```

**Comment by saw-jan** [comment]
waiting for PR https://github.com/owncloud/ocis/pull/8322, blocked until then

**Comment by saw-jan** [comment]
@nirajacharya2 Please, update the PR. the fix PR has been merged

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 4]
```diff
@@ -35,6 +35,8 @@ Feature: an user gets the resources shared to them
           "items": {
             "type": "object",
             "required": [
+              "@UI.Hidden",
+              "@client.synchronize",
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:39]: It will be done in https://github.com/owncloud/ocis/pull/8348

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 5]
```diff
@@ -532,3 +552,1116 @@ Feature: an user gets the resources shared to them
...
+    And the administrator has deleted user "Alice" using the provisioning API
+    When user "Brian" lists the shares shared with him using the Graph API
+    Then the HTTP status code should be "200"
+    """
+    {
+      "type": "object",
+      "required": [
+        "value"
+      ],
+      "properties": {
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:1407]: we have the share even the sharer is deleted?

CC @SagarGi

Comment (SagarGi) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature:1407]: no we should not have the response,  i have created an issue for it lets wait if its the actual bug> https://github.com/owncloud/ocis/issues/8356

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 6]
```diff
@@ -532,3 +552,1116 @@ Feature: an user gets the resources shared to them
...
+                  "testfile.txt"
+                ]
+              },
+              "parentReference": {
+                "type": "object",
+                "required": [
+                  "driveId",
+                  "driveType",
+                  "id"
+                ],
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:632]: for `parentReference` do we get like this in response https://github.com/owncloud/ocis/blob/4d5490a60135df8beb8a64626192e6555858f74e/tests/acceptance/features/apiSharingNg/sharedByMe.feature#L37-L45. may be we could also extend to it if we get them in the reponse. For other response also.

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 7]
```diff
@@ -1951,3 +1951,982 @@ Feature: an user gets the resources shared to them
       }
     }
     """
+
+
+    Scenario: user lists the file shared with them from project space
+    Given using spaces DAV path
+    Given the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:1958]:
```suggestion
    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 8]
```diff
@@ -1951,3 +1951,982 @@ Feature: an user gets the resources shared to them
...
+          }
+        }
+      }
+    }
+    """
+
+
+  Scenario: user lists the folder shared with them from project space
+    Given using spaces DAV path
+    Given the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:2209]:
```suggestion
    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 9]
```diff
@@ -1951,3 +1951,982 @@ Feature: an user gets the resources shared to them
...
+          }
+        }
+      }
+    }
+    """
+
+
+  Scenario: user lists the folder shared with them from project space after the sharer is disabled
+    Given using spaces DAV path
+    Given the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:2441]:
```suggestion
    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 10]
```diff
@@ -1951,3 +1951,982 @@ Feature: an user gets the resources shared to them
...
+          }
+        }
+      }
+    }
+    """
+
+  
+  Scenario: user lists the folder shared with them from project space after the sharer is deleted
+    Given using spaces DAV path
+    Given the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:2674]:
```suggestion
    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
```

**Review by amrita-shrestha — APPROVED** [review]
lgtm

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 11]
```diff
@@ -1951,3 +1951,982 @@ Feature: an user gets the resources shared to them
...
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+
+  
+  Scenario: user lists the folder shared with them from personal space after the sharer is deleted
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:2905]:
```suggestion
  Scenario: user lists the file shared with them from personal space after the sharer is deleted
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 12]
```diff
@@ -1951,3 +1951,982 @@ Feature: an user gets the resources shared to them
...
+                        "type": "object",
+                        "required": [
+                          "grantedToV2",
+                          "id",
+                          "invitation"
+                        ],
+                        "properties": {
+                          "id": {
+                            "type": "string"
+                          },
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:2116]: what value do we get here?

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 13]
```diff
@@ -2921,3 +2921,986 @@ Feature: an user gets the resources shared to them
...
+                              "user": {
+                                "type": "object",
+                                "properties": {
+                                  "displayName": {
+                                    "type": "string"
+                                  },
+                                  "id": {
+                                    "type": "string"
+                                  }
+                                },
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:3103]: If we are checking type then why not value. let's do it if possible

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature (hunk 5 above)
Comment (ScharfViktor) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature]: I see that it's correct behavior
personal space: sharer is deleted - no share anymore
project space: sharer is deleted - share is exist. Another user manager can manage this share

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 14]
```diff
@@ -3527,3 +3527,991 @@ Feature: an user gets the resources shared to them
...
+            }
+          }
+        }
+      }
+    }
+    """
+
+
+  Scenario: user lists the file shared with them from personal space after the sharer is deleted
+    Given the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4491]:
```suggestion

```

**Review by PrajwolAmatya — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 15]
```diff
@@ -3539,3 +3539,928 @@ Feature: an user gets the resources shared to them
...
+                                "user": {
+                                  "type": "object",
+                                  "required": [
+                                    "displayName",
+                                    "id"
+                                  ],
+                                  "properties": {
+                                    "displayName": {
+                                      "type": "string",
+                                      "enum": ["Alice Hansen"]
```
Comment (nabim777) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4406]:
```suggestion
                                      "enum": [""]
```
Since the user `Alice` is deleted but Alice is displayed. I think 2-3 second needed for user deletion.

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 16]
```diff
@@ -3539,3 +3539,928 @@ Feature: an user gets the resources shared to them
...
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created a space "new-space" with the default quota using the Graph API
+    And user "Alice" has created a folder "folder" in space "new-space"
+    And user "Alice" has sent the following share invitation:
+      | resource        | folder    |
+      | space           | new-space |
+      | sharee          | Brian     |
+      | shareType       | user      |
+      | permissionsRole | Viewer    |
+    And the administrator has deleted user "Alice" using the provisioning API
```
Comment (nabim777) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4235]:
```suggestion
    And the administrator has deleted user "Alice" using the provisioning API
```
Aren't we supposed to use Graph API instead of provisioning API?

Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4236]: I think, it's just the naming there. should use graph

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 17]
```diff
@@ -3539,3 +3539,929 @@ Feature: an user gets the resources shared to them
...
+              "lastModifiedDateTime",
+              "name",
+              "parentReference",
+              "remoteItem",
+              "size"
+            ],
+            "properties": {
+              "@UI.Hidden":{
+                "type": "boolean",
+                "enum": [false]
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:3586]: I am thinkgin of using `const` and not using `type`. two reasons: 1) less schema lines 2) const is best suited for exact comparison

`const` validator is supported now. previously, it wasn't

```suggestion
                "const": false
```

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 18]
```diff
@@ -2530,6 +2534,30 @@ public function userListsTheResourcesSharedWithThemUsingGraphApi(string $user):
...
+	/**
+	 * @When user :user lists the shares shared with him/her using the Graph API
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userListsTheResourcesSharedWithThemUsingGraphApi(string $user): void {
+		$this->getSharesSharedWithMe($user);
```
Comment (S-Panta) [inline tests/acceptance/features/bootstrap/GraphContext.php:2546]: Would it be better if the response is set with the variable in the `getSharesSharedWithMe`  and then these steps called that variable and set the response here? IMO that would be consistent with our code.

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 19]
```diff
@@ -3539,3 +3539,891 @@ Feature: an user gets the resources shared to them
...
+                "type": "string",
+                "pattern": "%etag_pattern%"
+              },
+              "file": {
+                "type": "object",
+                "required": ["mimeType"],
+                "properties": {
+                  "mimeType": {
+                    "type": "string",
+                    "pattern": "text/plain"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:3600]:
```suggestion
                    "type": "string",
                    "pattern": "^text/plain"
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 20]
```diff
@@ -3539,3 +3539,891 @@ Feature: an user gets the resources shared to them
...
+                            "type": "string",
+                            "pattern": "^%role_id_pattern%$"
+                          }
+                        }
+                      }
+                    }
+                  },
+                  "size": {
+                    "type": "number",
+                    "enum": [12]
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:3763]:
```suggestion
                    "const": 12
```

Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:3763]:
```suggestion
                    "const": 12,
```
Can't we use const here?

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 21]
```diff
@@ -3539,3 +3539,889 @@ Feature: an user gets the resources shared to them
...
+                      }
+                    }
+                  },
+                  "size": {
+                    "const": 12
+                  }
+                }
+              },
+              "size": {
+                "enum": 12
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:3767]:
```suggestion
                "const": 12
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 22]
```diff
@@ -3539,3 +3539,889 @@ Feature: an user gets the resources shared to them
...
+    And user "Alice" has created a space "new-space" with the default quota using the Graph API
+    And user "Alice" has created a folder "folder" in space "new-space"
+    And user "Alice" has sent the following share invitation:
+      | resource        | folder    |
+      | space           | new-space |
+      | sharee          | Brian     |
+      | shareType       | user      |
+      | permissionsRole | Viewer    |
+    And the user "Admin" has deleted a user "Alice"
+    When user "Brian" lists the shares shared with him after clearing user cache using the Graph API
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4205]: The step and scenario description do not match.

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 23]
```diff
@@ -3539,3 +3539,889 @@ Feature: an user gets the resources shared to them
...
+                  "driveId": {
+                    "type": "string",
+                    "pattern": "^%space_id_pattern%$"
+                  },
+                  "driveType" : {
+                    "const": "virtual"
+                  },
+                  "id" : {
+                    "type": "string",
+                    "pattern": "%space_id_pattern%"
```
Comment (nabim777) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4268]:
```suggestion
                    "pattern": "^%space_id_pattern%$"
```

Comment (nabim777) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4268]:
```suggestion
                    "pattern": "^%file_id_pattern%$"
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 24]
```diff
@@ -3539,3 +3539,889 @@ Feature: an user gets the resources shared to them
...
+                  "parentReference": {
+                    "type": "object",
+                    "required": [
+                      "driveId",
+                      "driveType"
+                    ],
+                    "properties": {
+                      "driveId": {
+                        "type": "string",
+                        "pattern": "%space_id_pattern%"
```
Comment (nabim777) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4305]:
```suggestion
                        "pattern": "^%space_id_pattern%$"
```

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 25]
```diff
@@ -2510,6 +2510,29 @@ public function userSwitchesTheSystemLanguageUsingGraphApi(string $user, string
...
+	 * @param bool $waitForUserDeletion
+	 *
+	 * @return ResponseInterface
+	 * @throws GuzzleException
+	 */
+	public function getSharesSharedWithMe(string $user, bool $waitForUserDeletion = false): ResponseInterface {
+		$credentials = $this->getAdminOrUserCredentials($user);
+		if ($waitForUserDeletion) {
+			// it takes few second to delete a user so wating until user is deleted
+			sleep(1);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/GraphContext.php:2525]: add some comments mentioning the env vars like in
https://github.com/owncloud/ocis/blob/f60e392d5977e45c760aea760a4756fd7671cfe2/tests/acceptance/features/bootstrap/GraphContext.php#L2542-L2547

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 26]
```diff
@@ -3539,3 +3539,879 @@ Feature: an user gets the resources shared to them
...
+                  "name",
+                  "parentReference",
+                  "permissions"
+                ],
+                "properties": {
+                  "eTag": {
+                    "type": "string",
+                    "pattern": "%etag_pattern%"
+                  },
+                  "folder": {},
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4073]: ?

Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4073]: how the json looks like?

Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4073]: then it should be:
```yml
"folder": {
    "const": {}
}
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 27]
```diff
@@ -3539,3 +3539,891 @@ Feature: an user gets the resources shared to them
...
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+
+  #
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:3983]:
```suggestion

```

**Review by grgprarup — APPROVED** [review]
LGTM :+1:

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 28]
```diff
@@ -3539,3 +3539,891 @@ Feature: an user gets the resources shared to them
...
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+
+  @env-config
+  Scenario: user lists the folder shared with them from project space after the sharer is deleted
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4194]: will be helpful
```suggestion
  @env-config @8314
  Scenario: user lists the folder shared with them from project space after the sharer is deleted
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 29]
```diff
@@ -3539,3 +3539,891 @@ Feature: an user gets the resources shared to them
...
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+
+
+  Scenario: user lists the file shared with them from personal space after the sharer is deleted
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:4405]: Please, add scenarios per item from https://github.com/owncloud/ocis/issues/7826 and the PR title would be meaningful.

**Review by saw-jan — APPROVED** [review]
